### DISCUSSION TITLE
Variable power at each timestep

### DISCUSSION THREAD

**Discussion by joshuaeh** [pybamm-team/PyBaMM#1989] [Q&A] [answered]
I am trying to use PyBaMM to create a digital twin. In doing so, I need to update the model at each minute time increment and get state of charge, voltage, state of health etc.

I tried to do so using the step function, however it seems that the `"Current function [A]"` taking the most recent value of the `net_power` array does not affect the model. Is this capability possible? Perhaps I am storing results wrong. Thanks for your help.

```
model = pybamm.lithium_ion.SPMe()

# create geometry
geometry = model.default_geometry

# load parameter values and process model and geometry
param = model.default_parameter_values
param.process_model(model)
param.process_geometry(geometry)

# set mesh
mesh = pybamm.Mesh(geometry, model.default_submesh_types, model.default_var_pts)

# discretise model
disc = pybamm.Discretisation(mesh, model.default_spatial_methods)
disc.process_model(model)

dt = 60
time = 3600*6
end_time = 3600 * 21
step_solver = pybamm.CasadiSolver()
step_solution = None
for i, a in enumerate(net_power):
    parameter_values = {}
    parameter_values["Current function [A]"] = a
    step_solution = step_solver.step(step_solution, model, dt=dt, npts=2, inputs=parameter_values)
```

**Comment by valentinsulzer** [comment, marked as answer]
You need to do `param["Current function [A]"] = "[input]"` before you do `param.process_model`. Then the rest of your script should work

**Reply by joshuaeh** [reply]
Thanks for your response @tinosulzer. I am trying to implement it now. Am I to understand that the exact line `param["Current function [A]"] = "[input]"` indicates that the Current function will be an input? Or do I need to rerun the line at each iteration: specifying the input, creating the model using `param.process_model`, and then stepping forward?

**Reply by valentinsulzer** [reply]
The former, you won't need to re-process the model at each iteration
